### PR TITLE
aws_transit_gateway_multicast_domain_association

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource. No integration tests with terraform are done as there is no terraform resource available.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by Nirbhay1997 — APPROVED** [review]

File: libraries/aws_transit_gateway_multicast_domain_associations.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+             .register_column(:subnets,                                         field: :subnet)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters(required: %i(transit_gateway_multicast_domain_id))
+    @query_params = {}
+    @query_params[:transit_gateway_multicast_domain_id] = opts[:transit_gateway_multicast_domain_id]
+    raise ArgumentError, "#{@__resource_name__}: transit_gateway_multicast_domain_id must be provided" unless opts[:transit_gateway_multicast_domain_id] && !opts[:transit_gateway_multicast_domain_id].empty?
+    @query_params[:transit_gateway_multicast_domain_id] = opts[:transit_gateway_multicast_domain_id]
```
Comment (sathish-progress) [inline libraries/aws_transit_gateway_multicast_domain_associations.rb:35]: this is duplicated

File: test/integration/verify/controls/aws_transit_gateway_multicast_domain_associations.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  describe aws_transit_gateway_multicast_domain_associations(transit_gateway_multicast_domain_id: "") do
+    it { should exist }
+  end
+
+  describe aws_transit_gateway_multicast_domain_associations(transit_gateway_multicast_domain_id: "") do
+    its('transit_gateway_attachment_ids') { should include '' }
+    its('resource_ids') { should include '' }
+    its('resource_types') { should include 'vpc' }
+    its('resource_owner_ids') { should include '' }
+  end
```
Comment (sathish-progress) [inline test/integration/verify/controls/aws_transit_gateway_multicast_domain_associations.rb:14]: why are we not testing anything here ?

Comment (soumyo13) [reply test/integration/verify/controls/aws_transit_gateway_multicast_domain_associations.rb:14]: I skipped it as the testing has to be done manually.

File: test/integration/verify/controls/aws_transit_gateway_multicast_domain_association.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  describe aws_transit_gateway_multicast_domain_association(transit_gateway_multicast_domain_id: '') do
+    it { should exist }
+  end
+
+  describe aws_transit_gateway_multicast_domain_association(transit_gateway_multicast_domain_id: "t") do
+    its('transit_gateway_attachment_id') { should eq '' }
+    its('resource_id') { should eq '' }
+    its('resource_type') { should eq 'vpc' }
+    its('resource_owner_id') { should eq '' }
+  end
```
[the remark this answers is not in the document]
Comment (soumyo13) [reply test/integration/verify/controls/aws_transit_gateway_multicast_domain_association.rb:14]: I skipped it as the testing has to be done manually.

File: libraries/aws_transit_gateway_multicast_domain_associations.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/aws_transit_gateway_multicast_domain_associations.rb:34]: Done and removed.

File: test/integration/verify/controls/aws_transit_gateway_multicast_domain_associations.rb (hunk 2 above)
Comment (soumyo13) [reply test/integration/verify/controls/aws_transit_gateway_multicast_domain_associations.rb]: Done with the dummy values

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**